src/puya/awst/wtypes.py

```diff
@@ -464,3 +465,46 @@ def is_arc4_argument_type(wtype: WType) -> bool:
         or is_reference_type(wtype)
         or (is_transaction_type(wtype) and wtype.transaction_type is not None)
     )
+
+
+def has_arc4_equivalent_type(wtype: WType) -> bool:
+    """
+    Checks if a non-arc4 encoded type has an arc4 equivalent
+    """
+    if wtype in (bool_wtype, uint64_wtype, bytes_wtype):
+        return True
+
+    match wtype:
+        case WTuple(types=types):
+            return all(has_arc4_equivalent_type(t) for t in types)
+    return False
+
+
+def avm_to_arc4_equivalent_type(wtype: WType) -> WType:
+    if wtype is bool_wtype:
+        return arc4_bool_wtype
+    if wtype is uint64_wtype:
```

Comment:
We could also support biguint_wtype here?

Reply:
biguint doesn't have a fixed precision so we would have to default to uint512 I guess which wouldn't be very efficient.

Reply:
I think allowing BigUInt and mapping to UInt512 makes sense for completeness.